contracts/SafeMinion.sol

```diff
@@ -0,0 +1,658 @@
+        require(!action.executed, ERROR_EXECUTED);
+        // Mark executed before doing any external stuff
+        actions[proposalId].executed = true;
+
+        if (action.memberOrModule) {
+            require(
+                isMember(msg.sender) || msg.sender == module,
+                ERROR_MEMBER_OR_MODULE_ONLY
+            );
+        }
+
+        require(isPassed(proposalId), ERROR_REQS_NOT_MET);
+        require(action.id != 0, ERROR_DELETED);
+        require(action.moloch == address(moloch), ERROR_MOLOCH_CHANGED);
+
+        bytes32 id = hashOperation(transactions);
+
+        require(id == action.id, ERROR_NOT_VALID);
+
+        if (
+            action.amount > 0 &&
+            moloch.getUserTokenBalance(address(executor), action.token) > 0
+        ) {
+            // withdraw token tribute if any
+            doWithdraw(
+                action.token,
+                moloch.getUserTokenBalance(address(executor), action.token)
+            );
+        }
+
+        require(
+            executor.execTransactionFromModule(
+                multisend,
+                0, // todo pull value?
+                transactions,
+                Operation.DelegateCall
+            ),
+            ERROR_CALL_FAIL
+        );
+
+        emit ExecuteAction(id, proposalId, transactions, msg.sender);
+
+        delete actions[proposalId];
+
+        return true;
+    }
+
+    function cancelAction(uint256 _proposalId) external {
+        Action memory action = actions[_proposalId];
+        require(msg.sender == action.proposer, ERROR_NOT_PROPOSER);
+        delete actions[_proposalId];
+        emit ActionCanceled(_proposalId);
+        moloch.cancelProposal(_proposalId);
+    }
+
+    // -- Admin Functions --
+    function changeOwner(address _moloch) external executorOnly returns (bool) {
+        // TODO: withdraw any funds from dao first? may need to verify this on the front end
+        moloch = IMOLOCH(_moloch);
+        molochDepositToken = moloch.depositToken();
+        // verify that moloch address has a deposit token and it is not zero
+        require(molochDepositToken != address(0), ERROR_ZERO_DEPOSIT_TOKEN);
+        emit ChangeOwner(_moloch);
+        return true;
+    }
+
+    function setModule(address _module) external executorOnly returns (bool) {
+        module = _module;
+        emit SetModule(_module);
+        return true;
+    }
+
+    //  -- Helper Functions --
+    function isPassed(uint256 _proposalId) internal returns (bool) {
+        uint256 totalShares = moloch.totalShares();
+        bool[6] memory flags = moloch.getProposalFlags(_proposalId);
+        require(flags[0], ERROR_NOT_SPONSORED);
+
+        // if any of these branches are true, let action proceed before proposal is processed
+        if (flags[2]) {
+            // if proposal has passed dao return true
+            return true;
+        }
+
+        if (module != address(0) && msg.sender == module) {
```

Comment:
This check also happens in `executeAction()` right before it calls `isPassed()`, so it's probably redundant here, no?